src/constants.js

```diff
@@ -14,16 +14,19 @@ export const namespaces = ['oci_computeagent', 'oci_blockstore', 'oci_lbaas', 'o
 export const aggregations = ['count()', 'max()', 'mean()', 'min()', 'rate()', 'sum()', 'percentile(.90)', 'percentile(.95)', 'percentile(.99)', 'last()']
 export const windows = [AUTO, '1m', '5m', '1h']
 export const resolutions = [AUTO, '1m', '5m', '1h']
-export const environments = ['local', 'OCI Instance']
-
-
-export const compartmentsQueryRegex = /^compartments\(\)\s*/;
-export const regionsQueryRegex = /^regions\(\)\s*/;
-export const namespacesQueryRegex = /^namespaces\(\s*(\".+\"|\'.+\'|\$\w+)\s*,\s*(\".+\"|\'.+\'|\$\w+)\s*\)/;
-export const resourcegroupsQueryRegex = /^resourcegroups\(\s*(\".+\"|\'.+\'|\$\w+)\s*,\s*(\".+\"|\'.+\'|\$\w+)\s*,\s*(\".+\"|\'.+\'|\$\w+)\s*\)/;
-export const metricsQueryRegex = /^metrics\(\s*(\".+\"|\'.+\'|\$\w+)\s*,\s*(\".+\"|\'.+\'|\$\w+)\s*,\s*(\".+\"|\'.+\'|\$\w+)\s*,\s*(\".+\"|\'.+\'|\$\w+)\s*\)/;
-export const dimensionKeysQueryRegex = /^dimensions\(\s*(\".+\"|\'.+\'|\$\w+)\s*,\s*(\".+\"|\'.+\'|\$\w+)\s*,\s*(\".+\"|\'.+\'|\$\w+)\s*,\s*(\".+\"|\'.+\'|\$\w+)\s*,\s*(\".+\"|\'.+\'|\$\w+)\s*\)/;
-export const dimensionValuesQueryRegex = /^dimensionOptions\(\s*(\".+\"|\'.+\'|\$\w+)\s*,\s*(\".+\"|\'.+\'|\$\w+)\s*,\s*(\".+\"|\'.+\'|\$\w+)\s*,\s*(\".+\"|\'.+\'|\$\w+)\s*,\s*(\".+\"|\'.+\'|\$\w+)\s*,\s*(\".+\"|\'.+\'|\$\w+)\s*\)/;
+export const environments = ['local', 'OCI Instance', 'multitenancy']
+
+
+// export const compartmentsQueryRegex = /^compartments\(\)\s*/;
+export const compartmentsQueryRegex = /^compartments\(\s*(\".+\"|\'.+\'|\$\w+)\s*,\s*(\".+\"|\'.+\'|\$\w+)\s*\)|^compartments\(\)\s*/;
```

Comment:
as we discussed yesterday, we need to pass one argument ($tenancy) to region, and two ($tenancy and $region) to compartment.

Reply:
However I think regions() should be used without argument. in the backend the tenancy is retrieved from the FE variable ts.Tenancyconfig, therefore I think during vars definition regions() should be defined without arguments. If passign an argument the plugin failed because is looking for compartments before populating the regions dropdown. I am still looking into this.

Reply:
For compartments() function having the region as a parameter in multi-tenancy mode is mandatory due to the fact that you may have your plugin configured to address different tenancies which resides in different region, so I need the two parameter to list the compartments. Without the region parameter I could not make it works